pkg/pillar/devicenetwork/dnc.go

```diff
@@ -728,7 +734,18 @@ func checkAndUpdateWireless(ctx *DeviceNetworkContext, oCfg *types.DevicePortCon
 				}
 			}
 		}
-		if oldPortCfg == nil || !reflect.DeepEqual(oldPortCfg.WirelessCfg, pCfg.WirelessCfg) {
+		if oldPortCfg == nil || strings.Contains(portCfg.Key, "override") ||
+			!reflect.DeepEqual(oldPortCfg.WirelessCfg, pCfg.WirelessCfg) {
+			// XXX hack to disable the cloud configure for wifi, if the existing configure
+			// has wifi information, but the new update does not. Remove this when the zedcloud
+			// implements the wifi configure for port
+			if oldPortCfg != nil && oldPortCfg.WirelessCfg.WType == types.WirelessTypeWifi &&
+				pCfg.WirelessCfg.WType == types.WirelessTypeNone {
+				log.Infof("checkAndUpdateWireless: HACK! wifi has configure %+v, ignore the update %+v\n",
```

Comment:
I wonder if a better hack would be to copy the existing WirelessCfg to pCfg and as a result pick up other changes to the portConfig from zedcloud.

Reply:
@eriknordmark since this function checkAndUpdateWireless() is only checking Wifi Config for handling wpa_supplicant.conf, it should still allow other part of the port config to take effect from zedcloud. Also I thought about copying over, the DPCModify would generate diff every time for the port old-config and new-config then keep updating.

Reply:
That's odd. Looking at the wireless code in doUpdatePortConfigListAndPublish it is all in the wrong place.
doUpdatePortConfigListAndPublish manipulates a list of entries, and does not commit to running any particular entry. Thus currently if a lower priority entry is added to the list, the code immediately changes the wireless config. That's wrong. Might explain the flapping Roman saw.

The application needs to be driven out of VerifyPending where the code tries a new portconfig.
Just before calling UpdateDhcpClient or something like that.

That way the wirelesscfg is properly tested before nim commits to using the a DevicePortConfig.

Reply:
Hey @naiming-zededa -- any chance you can take care of this today (as per @nordmark comments)? I'd like to cut a 4.5.1 (on account of a few bug fixes that seem pretty critical) and this one would be very nice to have.

Reply:
Hi @rvs yes, I'll look into this today.